hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodiePartitionCDCFileGroupMapping.scala

```diff
@@ -0,0 +1,118 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi
+
+import org.apache.hudi.common.model.HoodieFileGroupId
+import org.apache.hudi.common.table.cdc.HoodieCDCFileSplit
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.util.{ArrayData, MapData}
+import org.apache.spark.sql.types.{DataType, Decimal}
+import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
+
+import java.util
+
+case class HoodiePartitionCDCFileGroupMapping(partitionValues: InternalRow,
+                                              fileGroups: Map[HoodieFileGroupId, List[HoodieCDCFileSplit]]
+                                             ) extends InternalRow {
+
+  def getFileSplitsFor(fileGroupId: HoodieFileGroupId): Option[List[HoodieCDCFileSplit]] = {
+    fileGroups.get(fileGroupId)
+  }
+
+  override def numFields: Int = {
+    partitionValues.numFields
+  }
+
+  override def setNullAt(i: Int): Unit = {
+    partitionValues.setNullAt(i)
+  }
+
+  override def update(i: Int, value: Any): Unit = {
+    partitionValues.update(i, value)
+  }
+
+  override def copy(): InternalRow = {
+    HoodiePartitionCDCFileGroupMapping(partitionValues.copy(), fileGroups)
+  }
+
+  override def isNullAt(ordinal: Int): Boolean = {
+    partitionValues.isNullAt(ordinal)
+  }
+
+  override def getBoolean(ordinal: Int): Boolean = {
+    partitionValues.getBoolean(ordinal)
+  }
+
+  override def getByte(ordinal: Int): Byte = {
+    partitionValues.getByte(ordinal)
+  }
+
+  override def getShort(ordinal: Int): Short = {
+    partitionValues.getShort(ordinal)
+  }
+
+  override def getInt(ordinal: Int): Int = {
+    partitionValues.getInt(ordinal)
+  }
+
+  override def getLong(ordinal: Int): Long = {
+    partitionValues.getLong(ordinal)
+  }
+
+  override def getFloat(ordinal: Int): Float = {
+    partitionValues.getFloat(ordinal)
+  }
+
+  override def getDouble(ordinal: Int): Double = {
+    partitionValues.getDouble(ordinal)
+  }
+
+  override def getDecimal(ordinal: Int, precision: Int, scale: Int): Decimal = {
+    partitionValues.getDecimal(ordinal, precision, scale)
+  }
+
+  override def getUTF8String(ordinal: Int): UTF8String = {
+    partitionValues.getUTF8String(ordinal)
+  }
+
+  override def getBinary(ordinal: Int): Array[Byte] = {
+    partitionValues.getBinary(ordinal)
+  }
+
+  override def getInterval(ordinal: Int): CalendarInterval = {
+    partitionValues.getInterval(ordinal)
+  }
+
+  override def getStruct(ordinal: Int, numFields: Int): InternalRow = {
+    partitionValues.getStruct(ordinal, numFields)
+  }
+
+  override def getArray(ordinal: Int): ArrayData = {
+    partitionValues.getArray(ordinal)
+  }
+
+  override def getMap(ordinal: Int): MapData = {
+    partitionValues.getMap(ordinal)
+  }
+
+  override def get(ordinal: Int, dataType: DataType): AnyRef = {
+    partitionValues.getMap(ordinal)
```

Comment:
this should be `partitionValues.get(ordinal, dataType)`?

Reply:
Right. Let me fix.